packages/xdl/src/Webpack.js

```diff
@@ -336,3 +384,115 @@ async function getWebpackConfigEnvFromBundlingOptionsAsync(
     ...(options.webpackEnv || {}),
   };
 }
+
+async function startNextJsAsync({ projectRoot, port, dev, expoConfig, onFinished }) {
+  let next;
+  try {
+    next = require(path.join(projectRoot, 'node_modules', 'next'));
+  } catch {
+    throw new XDLError('NEXTJS_NOT_INSTALLED', 'Next.js is not installed in your app.');
+  }
+
+  // Build first if in production mode.
+  // https://nextjs.org/docs#custom-server-and-routing
+  if (!dev) {
+    await bundleNextJsAsync({ projectRoot, expoConfig });
+  }
+
+  const app = next({
+    dev,
+    dir: projectRoot,
+    conf: _createNextJsConf({
+      projectRoot,
+      expoConfig,
+    }),
+  });
+  const handle = app.getRequestHandler();
+
+  app.prepare().then(() => {
+    const server = express();
+    server.get('*', handle);
+
+    webpackDevServerInstance = server.listen(port, err => {
+      if (err) {
+        throw new XDLError(
+          'EXPRESS_FAIL_TO_START',
+          `Express server failed to start: ${err.toString()}`
+        );
+      }
+      onFinished();
+    });
+  });
+}
+
+async function bundleNextJsAsync({ projectRoot, expoConfig }) {
+  let nextBuild;
+  try {
+    nextBuild = require(path.join(projectRoot, 'node_modules', 'next', 'dist', 'build')).default;
+  } catch {
+    throw new XDLError(
+      'NEXTJS_NOT_INSTALLED',
+      'Next.js (or its build component) is not installed in your app.'
+    );
+  }
+
+  await nextBuild(projectRoot, _createNextJsConf({ projectRoot, expoConfig }));
+}
+
+function _createNextJsConf({ projectRoot, expoConfig }) {
+  let userNextConfigJs = {};
+  const userNextConfigJsPath = path.join(projectRoot, 'next.config.js');
+  if (fs.existsSync(userNextConfigJsPath)) {
+    userNextConfigJs = require(userNextConfigJsPath);
+  }
+
+  // `include` function is from https://github.com/expo/expo-cli/blob/3933f3d6ba65bffec2738ece71b62f2c284bd6e4/packages/webpack-config/webpack/loaders/createBabelLoaderAsync.js#L76-L96
+  const includeFunc = expoConfig.module.rules[1].include;
```

Comment:
Create a util method which attempts to find the loader by name and throws if it cannot be found.